src/OpenSSL/SSL.py

```diff
@@ -1486,7 +1488,13 @@ def sendall(self, buf, flags=0):
         data = _ffi.new("char[]", buf)
 
         while left_to_send:
-            result = _lib.SSL_write(self._ssl, data + total_sent, left_to_send)
```

Comment:
We constrain left_to_send with `min(left_to_send, 2147483647)` so it won't bail out. With this PR, pyOpenSSL will call `send()` multiple times (with smaller buffers) if a buffer >= 2**31 is passed to `sendall()`. Does that make sense?